kolibri/core/device/test/test_api.py

```diff
@@ -42,225 +36,13 @@
-
-    def test_dataset_set_created(self):
-        data = self._default_provision_data()
-        self._post_deviceprovision(data)
-        self.assertEqual(
-            FacilityDataset.objects.get().learner_can_edit_username,
-            self.dataset_data["learner_can_edit_username"],
-        )
-        self.assertEqual(
-            FacilityDataset.objects.get().learner_can_edit_name,
-            self.dataset_data["learner_can_edit_name"],
-        )
-        self.assertEqual(
-            FacilityDataset.objects.get().learner_can_edit_password,
-            self.dataset_data["learner_can_edit_password"],
-        )
-        self.assertEqual(
-            FacilityDataset.objects.get().learner_can_sign_up,
-            self.dataset_data["learner_can_sign_up"],
-        )
-        self.assertEqual(
-            FacilityDataset.objects.get().learner_can_delete_account,
-            self.dataset_data["learner_can_delete_account"],
-        )
-        self.assertEqual(
-            FacilityDataset.objects.get().learner_can_login_with_no_password,
-            self.dataset_data["learner_can_login_with_no_password"],
-        )
-
-    def test_device_settings_created(self):
-        data = self._default_provision_data()
-        self.assertEqual(DeviceSettings.objects.count(), 0)
-        self._post_deviceprovision(data)
-        self.assertEqual(DeviceSettings.objects.count(), 1)
-
-    def test_device_settings_values(self):
-        data = self._default_provision_data()
-        data["allow_guest_access"] = False
-        self._post_deviceprovision(data)
-        device_settings = DeviceSettings.objects.get()
-        self.assertEqual(device_settings.default_facility, Facility.objects.get())
-        self.assertFalse(device_settings.allow_guest_access)
-        self.assertFalse(device_settings.allow_peer_unlisted_channel_import)
-        self.assertTrue(device_settings.allow_learner_unassigned_resource_access)
-
-    def test_create_superuser_error(self):
-        data = self._default_provision_data()
-        data.update({"superuser": {}})
-        response = self._post_deviceprovision(data)
-        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-
-    def test_osuser_superuser_error_no_app(self):
-        with plugin_disabled("kolibri.plugins.app"):
-            data = self._default_provision_data()
-            del data["superuser"]
-            response = self._post_deviceprovision(data)
-            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-
-    def test_osuser_superuser_created(self):
-        with plugin_enabled("kolibri.plugins.app"), register_capabilities(
-            **{GET_OS_USER: lambda x: ("test_user", True)}
-        ):
-            initialize_url = interface.get_initialize_url(auth_token="test")
-            self.client.get(initialize_url)
-            data = self._default_provision_data()
-            del data["superuser"]
-            data.update({"auth_token": "test"})
```

Comment:
@rtibbles with the new task implementation, I am not handling this case. Should we do it?

Reply:
Done!